step-functions/oauth-token-generator.asl.json

```diff
@@ -1,29 +1,42 @@
 {
   "Comment": "HMRC OAuth Token Generator StateMachine",
-  "StartAt": "Get Totp Secret",
+  "StartAt": "Fetch HMRC Client Id",
   "States": {
-    "Get Totp Secret": {
+    "Fetch HMRC Client Id": {
       "Type": "Task",
-      "Next": "Lambda Invoke Generate TOTP Code",
+      "Next": "Fetch TOTP Secret",
+      "Parameters": {
+        "SecretId": "${ClientId}"
+      },
+      "Resource": "arn:aws:states:::aws-sdk:secretsmanager:getSecretValue",
+      "ResultPath": "$.clientId"
+    },
+    "Fetch TOTP Secret": {
+      "Type": "Task",
+      "Next": "Fetch HMRC Client Secret",
       "Parameters": {
         "SecretId": "${TotpSecretName}"
       },
       "Resource": "arn:aws:states:::aws-sdk:secretsmanager:getSecretValue",
-      "Catch": [
-        {
-          "ErrorEquals": ["States.ALL"],
-          "Next": "Error: Fetching TOTP Secret",
-          "Comment": "TOTP secret error"
-        }
-      ]
+      "ResultPath": "$.totpSecret"
     },
-    "Lambda Invoke Generate TOTP Code": {
+    "Fetch HMRC Client Secret": {
+      "Type": "Task",
+      "Next": "Generate TOTP Code",
+      "Parameters": {
+        "SecretId": "${ClientSecretName}"
```

Comment:
Please can you use selectors, so we don't get all the other garbage we don't care about filtered out, so we have just to data we need passed down from each state

Reply:
Is there a reason why we can put all the Fetch Secrets in a `parallel block`